src/manifests/input/input_manifest_1_0.py

```diff
@@ -0,0 +1,289 @@
+        return {
+            "schema-version": "1.0",
+            "build": self.build.__to_dict__(),
+            "ci": None if self.ci is None else self.ci.__to_dict__(),
+            "components": self.components.__to_dict__(),
+        }
+
+    def stable(self) -> 'InputManifest_1_0':
+        manifest: 'InputManifest_1_0' = copy.deepcopy(self)
+        manifest.components.__stabilize__()
+        return manifest
+
+    class Ci:
+        def __init__(self, data: dict) -> None:
+            self.image = None if data is None else self.Image(data.get("image", None))
+
+        def __to_dict__(self) -> Optional[dict]:
+            return None if self.image is None else {"image": self.image.__to_dict__()}
+
+        class Image:
+            def __init__(self, data: dict) -> None:
+                self.name = data["name"]
+                self.args = data.get("args", None)
+
+            def __to_dict__(self) -> dict:
+                return {
+                    "name": self.name,
+                    "args": self.args
+                }
+
+    class Build:
+        def __init__(self, data: dict) -> None:
+            self.name: str = data["name"]
+            self.version = data["version"]
+            self.qualifier = data.get("qualifier", None)
+            self.platform = data.get("platform", None)
+            self.architecture = data.get("architecture", None)
+            self.snapshot = data.get("snapshot", None)
+            self.patches = data.get("patches", [])
+
+        def __to_dict__(self) -> dict:
+            return {
+                "name": self.name,
+                "version": self.version,
+                "qualifier": self.qualifier,
+                "patches": self.patches,
+                "platform": self.platform,
+                "architecture": self.architecture,
+                "snapshot": self.snapshot,
+            }
+
+        @property
+        def filename(self) -> str:
+            return self.name.lower().replace(" ", "-")
+
+
+class InputComponents_1_0(Components['InputComponent_1_0 ']):
+    @classmethod
+    def __create__(self, data: dict) -> 'InputComponent_1_0':
+        return InputComponent_1_0._from(data)  # type: ignore[no-any-return]
+
+    def __stabilize__(self) -> None:
+        for component in self.values():
+            component.__stabilize__()
+
+    def select(self, focus: List[str] = [], platform: str = None) -> Iterator['InputComponent_1_0']:
+        """
+        Select components.
+
+        :param List[str] focus: Choose some components.
+        :param str platform: Only components targeting a given platform.
+        :return: Collection of components.
+        :raises ValueError: Invalid platform or component name specified.
+        """
+        if focus and len(focus) > 0:
+            invalid = [item for item in focus if item not in self]
+            if len(invalid) > 0:
+                raise ValueError(f"Unknown component{'s'[:len(invalid) != 1]}={','.join(invalid)}.")
```

Comment:
please add missing tests.

Reply:
These functions are basically the same as the main `input_manifest` class, just the previous `1.0` schema version. However I did add some tests for different schema versions in `tests/tests_manifests/test_input_manifest.py`. 
I don't see we have separate tests for old schema version `build_manifest` or `bundle_manifest`. Do we want to create duplicate tests for this class especially?